src/components/day/Day.js

```diff
@@ -317,13 +321,39 @@ export default class DayComponent extends Field {
     }
   }
 
+  normalizeValue(value) {
+    if (!value || this.valueMask.test(value)) {
+      return value;
+    }
+    const dateParts = [];
+    const valueParts = value.split('/');
+
+    const getNextPart = (shouldTake, defaultValue) =>
+      dateParts.push(shouldTake ? valueParts.shift() : defaultValue);
+
+    if (this.dayFirst) {
+      getNextPart(this.showDay, '00');
+    }
+
+    getNextPart(this.showMonth, '00');
+
+    if (!this.dayFirst) {
+      getNextPart(this.showDay, '00');
+    }
+
+    getNextPart(this.showYear, '0000');
+
+    return dateParts.join('/');
+  }
+
   /**
    * Set the value at a specific index.
    *
    * @param index
    * @param value
    */
   setValueAt(index, value) {
+    value = this.normalizeValue(value);
```

Comment:
This should be getting called in ```updateValue``` which is executed before it gets to this point. It should also handle multiple values with multiple flag which this is getting called within each value. This means that your implementation changes behavior of how the normal normalizeValue should work.  This may need some work.

Reply:
@travist are you sure that Day Component may resolve multiple values?